src/compiler/types.ts

```diff
@@ -1946,6 +1946,7 @@ namespace ts {
         UseFullyQualifiedType           = 0x00000080,  // Write out the fully qualified type name (eg. Module.Type, instead of Type)
         InFirstTypeArgument             = 0x00000100,  // Writing first type argument of the instantiated type
         InTypeAlias                     = 0x00000200,  // Writing type in type alias declaration
+        UseTypeAliasValue               = 0x00000400,  // Writing infered type alias as its value. This is used when we emit declaration file

```

Comment:
how about `If there is a type alias declaration matching the type use it instead of serializing the type.`